plugins/router/template/plugin.go

```diff
@@ -109,18 +109,14 @@ func (p *TemplatePlugin) HandleRoute(eventType watch.EventType, route *routeapi.
 	return p.Router.Commit()
 }
 
-// TODO: the internal keys for routes and endpoints should be namespaced.  Currently
-// there is an upstream issue where the namespace is not set on non-resolved endpoints.
-// A fix has been submitted and we should consume it in the next rebase.
-
 // routeKey returns the internal router key to use for the given Route.
 func routeKey(route routeapi.Route) string {
-	return route.ServiceName
+	return fmt.Sprintf("%s-%s", route.Namespace, route.ServiceName)
```

Comment:
No problem.  Since the key is also used inside the template as part of the name for the backend I will also include a method to provide a template safe version.  I need this for path based routing anyway (working on it today) so I'm glad it came up.